infrastructure/iam_for_lambda.tf

```diff
@@ -1,5 +1,51 @@
-resource "aws_iam_role" "nfish-des-role-lambda_s3_access" {
-  name = "nfish-des-role-lambda_s3_access"
+resource "aws_iam_policy" "nfish-des-pol-lambda_s3_dynamodb" {
+  name        = "nfish-des-pol-lambda_s3_dynamodb"
+  description = "access to s3 and dynamodb for lambda"
+  path        = "/"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3-object-lambda:Get*",
+                "s3-object-lambda:PutObject",
+                "s3-object-lambda:DeleteObject"
+            ],
+            "Resource": "arn:aws:s3:::nfish-des-kutter-photos"
```

Comment:
much better. 
If you want to reference the resource directly you can also use 
[iam_policy_document](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/iam_policy_document)